app/models/spree/price_decorator.rb

```diff
@@ -0,0 +1,11 @@
+Spree::Price.class_eval do
+  
+  def price_including_vat_for(price_options)
+    options = price_options.merge(tax_category: variant.tax_category)
+    if variant.sale_price.present?
```

Comment:
@rishabhguptarishi it will be better to create a method in variant decorator which returns sale price if present else returns price rather than repeating that condition everywhere.